Uplift/Views/DropdownView.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  DropdownView.swift
+//  
+//
+//  Created by Cameron Hamidi on 10/7/19.
+//
+
+import Foundation
+import UIKit
+import SnapKit
+
+enum DropdownStatus {
+    case closed, half, open
+}
+
+enum DropdownDirection {
+    case horizontal, vertical
+}
+
+protocol DropdownViewDelegate: class {
+    func dropdownViewClosed(sender dropdownView: DropdownView)
+    func dropdownViewOpen(sender dropdownView: DropdownView)
+    func dropdownViewHalf(sender dropdownView: DropdownView)
+}
+
+class DropdownView: UIView {
+    
+    // MARK: - INITIALIZATION
+    private var contentView: UIView!
```

Comment:
`private let contentView = UIView()`

Reply:
`contentView` is supplied as an argument by whomever initializes the `Dropdown`view so it can't be a constant

Reply:
If you move the assignments to before the call to super.init(), the fields can be constant (tried it and it built)

Reply:
Should these even be constant though? What if the user decides to change the content of the dropdown view down the line?

Reply:
The content can still change, you're changing them accessing contentView's properties. What we know shouldn't change is the contentView itself (there should only be one and we shouldn't be creating another one with new properties to reassign to this variable when we can just change the properties of this variable)